File: 5회차-linearDS/5회차 필수문제 코드/CPP/1863.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;


int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n;
    cin >> n;

    stack<int> stk;
    int ans = 0;

    for (int i = 0; i < n; ++i) {
        int a, b;
        cin >> a >> b;

        if (stk.empty() || stk.top() < b) 
            stk.push(b);
        else if (stk.top() != b) {
            while (!stk.empty() && stk.top() > b) {
                stk.pop();
                ans++;
            }

            if (stk.empty() || stk.top() < b)
                stk.push(b);
        }
    }
    
    while (!stk.empty()) {
        if (stk.top() != 0)
            ans++;
        stk.pop();
    }
    cout << ans << '\n';
}

File: 5회차-linearDS/5회차 필수문제 코드/PYTHON/1863.py
n = int(input())
stk = []
ans = 0

for i in range(n):
    a, b = map(int, input().split())

    if len(stk) == 0 or stk[-1] < b:
        stk.append(b)
    elif stk[-1] != b:
        while len(stk) != 0 and stk[-1] > b:
            stk.pop()
            ans += 1

        if len(stk) == 0 or stk[-1] < b:
            stk.append(b)

while len(stk) != 0:
    if stk[-1] != 0:
        ans += 1
    stk.pop()

print(ans)